algorithms/src/snark/varuna/ahp/selectors.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright (C) 2019-2023 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::fft::EvaluationDomain;
+use snarkvm_fields::{batch_inversion, PrimeField};
+use snarkvm_utilities::{cfg_into_iter, serialize::*};
+
+use itertools::Itertools;
+use std::collections::{BTreeMap, HashSet};
+
+#[cfg(not(feature = "serial"))]
+use rayon::prelude::*;
+
+use super::verifier::QueryPoints;
+
+/// Given two domains H and K such that H \subseteq K,
+/// construct polynomial that outputs 0 on all elements in K \ H, but 1 on all elements of H.
+pub(crate) trait SelectorPolynomial<F: PrimeField> {
+    fn evaluate_selector_polynomial(&self, other: EvaluationDomain<F>, point: F) -> F;
+}
+
+impl<F: PrimeField> SelectorPolynomial<F> for EvaluationDomain<F> {
+    fn evaluate_selector_polynomial(&self, other: EvaluationDomain<F>, point: F) -> F {
+        let numerator = self.evaluate_vanishing_polynomial(point) * other.size_as_field_element;
+        let denominator = other.evaluate_vanishing_polynomial(point) * self.size_as_field_element;
+        numerator / denominator
+    }
+}
```

Comment:
I'd recommend moving this to `EvaluationDomain`, so that we don't need a separate trait for it.

Reply:
Good idea. Actually its not used anymore so I moved it into the test, also moved the other free selector function here: 7bd8afc1f